catppuccin-theme.el

```diff
@@ -30,846 +32,934 @@ The theme has to be reloaded after changing anything in this group."
+               (whitespace-big-indent :foreground ,ctp-peach)
+               (whitespace-empty :inherit warning)
+               (whitespace-hspace :background ,undef :foreground ,undef)
+               (whitespace-indentation :foreground ,ctp-surface0)
+               (whitespace-line :underline (:style wave :color ,ctp-mauve))
+               (whitespace-newline :inherit font-lock-comment-face)
+               (whitespace-space :inherit font-lock-comment-face)
+               (whitespace-space-after-tab :inherit warning)
+               (whitespace-space-before-tab :inherit warning)
+               (whitespace-tab :inherit whitespace-newline)
                (whitespace-trailing :inherit trailing-whitespace)
                ;; yard-mode
                (yard-tag-face :inherit font-lock-builtin-face)
                (yard-directive-face :inherit font-lock-builtin-face))))
 
   (apply #'custom-theme-set-faces
          'catppuccin
-         (let ((expand-with-func
-                (lambda (func spec)
-                  (let (reduced-color-list)
-                    (dolist (col colors reduced-color-list)
-                      (push (list (car col) (funcall func col))
-                            reduced-color-list))
-                    (eval `(let ,reduced-color-list
-                             (backquote ,spec))))))
-               whole-theme)
-           (pcase-dolist (`(,face . ,spec) faces)
-             (push `(,face
-                     ((((min-colors 16777216)) ; fully graphical envs
-                       ,(funcall expand-with-func 'cadr spec))
-                      (((min-colors 256))      ; terminal withs 256 colors
-                       ,(if catppuccin-use-24-bit-colors-on-256-colors-terms
-                            (funcall expand-with-func 'cadr spec)
-                          (funcall expand-with-func 'caddr spec)))
-                      (t                       ; should be only tty-like envs
-                       ,(funcall expand-with-func 'cadddr spec))))
-                   whole-theme))
-           whole-theme))
+         (let* ((expand-with-func
+                 (lambda (func spec)
+                   (let (reduced-color-list)
+                     (dolist (col colors reduced-color-list)
+                       (push (list (car col) (funcall func col))
+                             reduced-color-list))
+                     (eval `(let ,reduced-color-list
+                              (backquote ,spec))))))
+                whole-theme)
+            (pcase-dolist (`(,face . ,spec) faces)
+              (push `(,face
+                      ((((min-colors 16777216)) ; fully graphical envs
+                        ,(funcall expand-with-func 'cadr spec))
+                       (t                       ; terminal with 256 colors
+                        ,(funcall expand-with-func 'caddr spec))))
+                    whole-theme))
+            whole-theme))
 
   (apply #'custom-theme-set-variables
```

Comment:
There should be added something here probably about `rustic-ansi-faces` to make the rustic terminal use catppuccin colors